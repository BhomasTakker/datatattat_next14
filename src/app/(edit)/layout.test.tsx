import React from "react";
import { render, screen } from "@testing-library/react";
import RootLayout from "./layout";

const TestChild = () => <div data-testid="test-child">Child Content</div>;
const params = Promise.resolve({ route: [] });

// Mock the imported components
jest.mock("../../components/header/main-header", () => ({
	MainHeader: () => <div data-testid="main-header" />,
}));
jest.mock("../../components/ui/layout/page-layout", () => ({
	PageLayout: ({ children }: { children: React.ReactNode }) => (
		<div data-testid="page-layout">{children}</div>
	),
}));
jest.mock("../../lib/sonner/toast-container", () => ({
	ToastContainer: () => <div data-testid="toast-container" />,
}));

describe("RootLayout", () => {
	it("renders PageLayout, MainHeader, children, and ToastContainer", async () => {
		render(await RootLayout({ children: <TestChild />, params }));

		expect(screen.getByTestId("page-layout")).toBeInTheDocument();
		expect(screen.getByTestId("main-header")).toBeInTheDocument();
		expect(screen.getByTestId("test-child")).toBeInTheDocument();
		expect(screen.getByTestId("toast-container")).toBeInTheDocument();
	});

	describe("Snaps", () => {
		it("renders Layout Unchanged", async () => {
			const { container } = render(
				await RootLayout({ children: <TestChild />, params })
			);
			expect(container).toMatchSnapshot();
		});
	});
});
