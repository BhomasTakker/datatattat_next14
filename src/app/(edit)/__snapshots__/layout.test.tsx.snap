// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RootLayout Snaps renders Layout Unchanged 1`] = `
<div>
  <div
    data-testid="page-layout"
  >
    <div
      data-testid="main-header"
    />
    <div
      data-testid="test-child"
    >
      Child Content
    </div>
    <div
      data-testid="toast-container"
    />
  </div>
</div>
`;
